packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -85,71 +91,63 @@ async function proxyPreviewModeApiRouteHandler(
     } catch {}
   }
 
-  previewModeProxy.once('proxyReq', proxyReq => {
-    proxyReq.removeHeader('X-Makeswift-Preview-Mode')
-
-    // The following headers are Next.js-specific and are removed to prevent Next.js from
-    // short-circuiting requests, breaking routing.
-    proxyReq.removeHeader('X-Invoke-Path')
-    proxyReq.removeHeader('X-Invoke-Query')
```

Comment:
We seem to be losing some context here as to why these headers were being removed. As we move things around we should try and maintain this information.